src/pyfaf/actions/releasedel.py

```diff
@@ -0,0 +1,195 @@
+        query.delete(False)
+
+        # delete all builds and packages from them
+        for build in all_builds:
+            for pkg in (db.session.query(st.Package)
+                        .filter(st.Package.build_id == build.build_id)
+                        .all()):
+                self._delete_package(pkg)
+
+    def _delete_release_repos(self, db, opsysrelease_id):
+        self.log_info("Removing repositories")
+        all_oprelrepos = (db.session.query(st.OpSysReleaseRepo)
+                          .filter(st.OpSysReleaseRepo.opsysrelease_id == opsysrelease_id)
+                          .all())
+        self.log_info("Repositories found: {0}".format(len(all_oprelrepos)))
+
+        for oprelrepo in all_oprelrepos:
+            repo = (db.session.query(st.Repo)
+                    .filter(st.Repo.id == oprelrepo.repo_id)
+                    .first())
+
+            self.log_debug("Removing repository '{0}'".format(repo.name))
+
+            for url in repo.url_list:
+                db.session.delete(url)
+
+            db.session.delete(repo)
+
+    def _delete_release_reports(self, db, opsysrelease_id):
+        self.log_info("Removing reports")
+        (db.session.query(st.ReportReleaseDesktop)
+         .filter(st.ReportReleaseDesktop.release_id == opsysrelease_id)
+         .delete(False))
+
+        (db.session.query(st.ReportHistoryDaily)
+         .filter(st.ReportHistoryDaily.opsysrelease_id == opsysrelease_id)
+         .delete(False))
+
+        (db.session.query(st.ReportHistoryWeekly)
+         .filter(st.ReportHistoryWeekly.opsysrelease_id == opsysrelease_id)
+         .delete(False))
+
+        (db.session.query(st.ReportHistoryMonthly)
+         .filter(st.ReportHistoryMonthly.opsysrelease_id == opsysrelease_id)
+         .delete(False))
+
+        (db.session.query(st.ReportOpSysRelease)
+         .filter(st.ReportOpSysRelease.opsysrelease_id == opsysrelease_id)
+         .delete(False))
+
+        # find and delete all reports, that are not assigned to any opsysreleases
+        report_query = (db.session.query(st.Report)
+                        .filter(~db.session.query()
+                                .exists()
+                                .where(st.ReportOpSysRelease.report_id == st.Report.id)))
+
+        self.log_info("Reports found: {0}".format(report_query.count()))
+        report_query.delete(False)
+
+    def _delete_release_problems(self, db, opsysrelease_id):
+        self.log_info("Removing problems")
+        (db.session.query(st.ProblemOpSysRelease)
+         .filter(st.ProblemOpSysRelease.opsysrelease_id == opsysrelease_id)
+         .delete(False))
+
+        empty_problems = get_empty_problems(db)
+        self.log_info("Empty problems found: {0}".format(len(empty_problems)))
+
+        for problem in empty_problems:
+            self.log_debug("Removing empty problem #{0}".format(problem.id))
+            db.session.delete(problem)
+
+    def _delete_package(self, pkg):
```

Comment:
Can you merge it with src/pyfaf/actions/cleanup_unassigned.py:delete_package() ?